src/dataflow/src/render/reduce.rs

```diff
@@ -172,19 +180,27 @@ where
             } else {
                 // Collect aggregates with their indexes, so they can be sliced and diced.
                 let mut accumulable = Vec::new();
+                let mut min_or_max = Vec::new();
                 let mut remaining = Vec::new();
                 for index in 0..aggregates.len() {
                     if accumulable_hierarchical(&aggregates[index].func).0 {
                         accumulable.push((index, aggregates[index].clone()));
+                    } else if is_min_or_max(&aggregates[index].func) && !monotonic {
```

Comment:
structuring this specifically for min / max + non-monotonic is obnoxious and deserves either a fix or a comment

Reply:
added a comment which will have to do for now

Reply:
To record a conversation yesterday: monotonic mins and maxs could be handled by the accumulable dataflow; they are inserted in to the diff field anyhow, and in the case of monotonicity we only need the one `reduce` operator for any number of sums and mins/maxs.

That seems like the ideal dataflow shape in that case, minimizing the number of arrangements, but no information about whether it is more or less complicated to express.

Reply:
So my opinion from yesterday's conversation (not strongly held) is that even though monotonic min / max are accumulable in the way you've described, I'm not sure if its worth the code complexity to try to fuse them together with the accumulable reductions in that case, and I would instead prefer to structure the code to have a `build_accumulables` function that generates one dataflow fragment for all accumulable reductions, and one `build_min_max` function that generates one dataflow fragment for all mins and maxes in all cases. Then, we could have a function called `build_single_aggregate` to catch the case of something like `jsonb_agg`

If at a later date, someone is combining accumulable reductions and monotonic min/max in a single view and wants some memory optimization, I think we can try to fold the two into a single arrangement then but for now its unclear to me if its necessary.

But like I said, this opinion is weakly held :)